dependencies.yaml

```diff
@@ -165,11 +165,19 @@ dependencies:
         packages:
           - &cmake_ver cmake>=3.30.4,<4
           - ninja
+  clang_tidy:
+    common:
+      - output_types: conda
+        packages:
+          - &clang_ver clang==20.1.4
+          - &clang_tools_ver clang-tools==20.1.4
   build-cpp:
     common:
       - output_types: conda
         packages:
           - c-compiler
+          - *clang_ver
+          - *clang_tools_ver
```

Comment:
I thought some more after approving. Are these needed? Why? The `clang_tidy` list is already in the "all" environment so these should be included in developer environments. I don't see a reason to duplicate them here.